release_util/management/commands/show_unapplied_migrations.py

```diff
@@ -2,50 +2,88 @@
+    2) To detect if any unapplied migrations exist and failing if so (by specifying '--fail_on_unapplied')
     """
-    help = "Prints out a JSON string of any unapplied migrations, along with their accompanying application name."
+    help = "Prints out a YAML string of any unapplied migrations, along with their accompanying application name."
 
-    def handle(self, *args, **kwargs):
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--fail_on_unapplied',
+            dest='fail_on_unapplied',
+            action='store_true',
+            help="If flag specified, command will exit with a non-zero value when unapplied migrations exist.",
+        )
+        parser.add_argument(
+            '--database',
+            dest='database',
+            default=DEFAULT_DB_ALIAS,
+            help='Nominates a database to synchronize. Defaults to the "default" database.',
+        )
+        parser.add_argument(
+            '--output_file',
+            dest='output_file',
+            default=None,
+            help="Filename to which output should be written."
+        )
 
-        unapplied = []
+    def handle(self, *args, **kwargs):
 
+        unapplied = defaultdict(list)
         out = StringIO()
-        ansi_escape = re.compile(r'\x1b[^m]*m')
-        for db in settings.DATABASES.keys():
-
-            try:
-                call_command("showmigrations", database=db, stdout=out, format="list")
-            except OperationalError:
-                self.stderr.write("Unable to check migrations: cannot connect to database '{}'.\n".format(db))
-                self.stdout.write("null")
-                sys.exit(1)
-
-            for line in out.getvalue().split('\n'):
-                line = ansi_escape.sub('', line)
-                if not line.startswith(' '):
-                    # This is an application line.
-                    current_app = line.strip()
-                elif '[ ]' in line:
-                    unapplied.append((current_app, line.strip().replace('[ ] ', '')))
-
-        if unapplied:
-            self.stdout.write(json.dumps(unapplied))
+
+        # This implementation is mostly taken from the Django 'showmigrations' mgmt command.
+        db = kwargs['database']
+        connection = connections[db]
+        loader = MigrationLoader(connection, ignore_no_migrations=True)
+        graph = loader.graph
+        app_names = sorted(loader.migrated_apps)
+        # For each app, print its migrations in order from oldest (roots) to
+        # newest (leaves).
+        for app_name in app_names:
+            shown = set()
+            for node in graph.leaf_nodes(app_name):
+                for plan_node in graph.forwards_plan(node):
+                    if plan_node not in shown and plan_node[0] == app_name:
+                        if not plan_node in loader.applied_migrations:
+                            unapplied[app_name].append(plan_node[1])
+                        shown.add(plan_node)
```

Comment:
Pulling this out in to it's own function might make the code easier to test. Then you could have another function that would produce the yaml using the output from this new method. They seem like two different concepts we should be able to test independently.